api/models/node.go

```diff
@@ -82,6 +82,46 @@ func (t IndexTags) Trace(span opentracing.Span) {
 func (i IndexTags) TraceDebug(span opentracing.Span) {
 }
 
+type IndexAutoCompleteTags struct {
+	OrgId     int      `json:"orgId" binding:"Required"`
+	TagPrefix string   `json:"tagPrefix"`
+	Expr      []string `json:"expressions"`
+	From      int64    `json:"from"`
+	Limit     uint16   `json:"limit"`
+}
+
+func (t IndexAutoCompleteTags) Trace(span opentracing.Span) {
+	span.SetTag("org", t.OrgId)
+	span.SetTag("tagPrefix", t.TagPrefix)
+	span.SetTag("expressions", t.Expr)
+	span.SetTag("from", t.From)
+	span.SetTag("limit", t.Limit)
+}
+
+func (i IndexAutoCompleteTags) TraceDebug(span opentracing.Span) {
+}
+
+type IndexAutoCompleteTagValues struct {
+	OrgId     int      `json:"orgId" binding:"Required"`
+	ValPrefix string   `json:"valPrefix"`
+	Tag       string   `json:"tag"`
+	Expr      []string `json:"expressions"`
+	From      int64    `json:"from"`
+	Limit     uint16   `json:"limit"`
```

Comment:
I believe these default to `100` in graphite.

Reply:
Yeah, in graphite the default is `100` if no limit value is given. If the given limit value is `0` then this will be accepted as a valid value.
In Macaron's request binding I can't see an easy way to make it behave exactly the same way because if no value is given it will be set to `0` and there's no `None`. But I cannot think of a case where somebody would actually want the limit to be `0` so I'd suggest that we'll just update `0` to a configurable default value (like in graphite).

Reply:
https://github.com/grafana/metrictank/pull/779/commits/c0da443455b40549eb685f3492b595072091aa14

Reply:
Could you use `binding:"Default(100)"`?

Reply:
Ah right, thx i missed that somehow. Although with that the problem is that afaict we can't make the default value configurable anymore. 
The solution I was looking at was via the `Validate()` method, but in that method the information whether a field was not set or if it was set to `0` is already lost.
Not sure what the preferred behavior is in that situation, I think making the default configurable is good and we want to keep that.